econml/iv/dr/_dr.py

```diff
@@ -367,78 +1680,126 @@ class _IntentToTreatDRIVModelNuisance:
 
-    def predict(self, Y, T, X=None, W=None, Z=None, sample_weight=None):
-        Y_pred = self._model_Y_X.predict(X, W)
-        T_pred_zero = self._model_T_XZ.predict(X, W, np.zeros(Z.shape))
-        T_pred_one = self._model_T_XZ.predict(X, W, np.ones(Z.shape))
-        delta = (T_pred_one - T_pred_zero) / 2
-        T_pred_mean = (T_pred_one + T_pred_zero) / 2
+    def predict(self, Y, T, X=None, W=None, Z=None, sample_weight=None, groups=None):
+        Y_pred = self._model_y_xw.predict(X, W)
+        T_pred_zero = self._model_t_xwz.predict(X, self._combine(W, np.zeros(Z.shape), Y.shape[0]))
+        T_pred_one = self._model_t_xwz.predict(X, self._combine(W, np.ones(Z.shape), Y.shape[0]))
+        Z_pred = self._dummy_z.predict(X, W)
         prel_theta = self._prel_model_effect.effect(X)
+
         if X is None:  # In this case predict above returns a single row
-            Y_pred = np.tile(Y_pred.reshape(1, -1), (Y.shape[0], 1))
             prel_theta = np.tile(prel_theta.reshape(1, -1), (T.shape[0], 1))
-        Y_res = Y - Y_pred.reshape(Y.shape)
-        T_res = T - T_pred_mean.reshape(T.shape)
-        return prel_theta, T_res, Y_res, 2 * Z - 1, delta
+            if W is None:
+                Y_pred = np.tile(Y_pred.reshape(1, -1), (Y.shape[0], 1))
+                Z_pred = np.tile(Z_pred.reshape(1, -1), (Z.shape[0], 1))
+
+        # reshape the predictions
+        Y_pred = Y_pred.reshape(Y.shape)
+        Z_pred = Z_pred.reshape(Z.shape)
+        T_pred_one = T_pred_one.reshape(T.shape)
+        T_pred_zero = T_pred_zero.reshape(T.shape)
+
+        # T_res, Z_res, beta expect shape to be (n,1)
+        beta = Z_pred * (1 - Z_pred) * (T_pred_one - T_pred_zero)
+        T_pred = T_pred_one * Z_pred + T_pred_zero * (1 - Z_pred)
+        Y_res = Y - Y_pred
+        T_res = T - T_pred
+        Z_res = Z - Z_pred
+
+        return prel_theta, Y_res, T_res, Z_res, beta
+
+
+class _DummyClassifier:
+    """
+    A dummy classifier that always returns the prior ratio
+
+    Parameters
+    ----------
+    ratio: float
+        The ratio of treatment samples
+    """
+
+    def __init__(self, *, ratio):
+        self.ratio = ratio
+
+    def fit(self, X, y, **kwargs):
+        return self
+
+    def predict_proba(self, X):
+        n = X.shape[0]
+        return np.hstack((np.tile(1 - self.ratio, (n, 1)), np.tile(self.ratio, (n, 1))))
 
 
 class _IntentToTreatDRIV(_BaseDRIV):
     """
-    Helper class for the DRIV algorithm for the intent-to-treat A/B test setting
+    Base class for the DRIV algorithm for the intent-to-treat A/B test setting
     """
 
     def __init__(self, *,
-                 model_Y_X,
-                 model_T_XZ,
+                 model_y_xw="auto",
+                 model_t_xwz="auto",
                  prel_model_effect,
                  model_final,
+                 z_propensity="auto",
                  featurizer=None,
-                 fit_cate_intercept=True,
+                 fit_cate_intercept=False,
```

Comment:
it comes from _BaseDRIV._BaseDRIV is the base class for all DRIV and IntentToTreatDRIV classes, includes Linear and non linear final stage. Only LinearDRIV and LinearIntentToTreatDRIV they do have fit_cate_intercept, others it has to be False (since we don't have the concecpt of intercept), so for an internal base class I set it as False as default.